### PR TITLE
kete: various SVG export improvements

### PR DESCRIPTION
**Pull request by mont5piques**
Read device configuration from {CONFIG_PATH}/settings.ini
Read tablet orientation from config
Handle tablet pressure field by adapting stroke width (activated thru
config)
Normalize SVG dimensions (increase viewer compatibility)

### CODE REVIEW

File: tools/kete.py [hunk 1]
```diff
@@ -543,13 +543,19 @@ def _log_drawings_available(self, device):
...
         super(Fetcher, self).__init__(manager)
         self.device = None
         self.timestamps = None
         address = args.address
         index = args.index
 
+        if not address in config:
+            config[address] = {}
+
+        self.orientation = int(config[address].get('Orientation', 0))
```
Comment (bentiss) [inline tools/kete.py]: Can we make this a string instead?
Like `Portrait`, `Landscape`, `Upside-Down-Portrait`, `Upside-Down-Landscape` (not usre about how to properly name the last 2).

File: tools/kete.py [hunk 2]
```diff
@@ -543,13 +543,19 @@ def _log_drawings_available(self, device):
...
+        self.handle_pressure = int(config[address].get('HandlePressure', 0))
```
Comment (bentiss) [inline tools/kete.py]: Ideally I would prefer this to be a boolean, and use `config[address].getboolean('HandlePressure', False)` instead

File: tools/kete.py [hunk 3]
```diff
@@ -543,13 +543,19 @@ def _log_drawings_available(self, device):
...
-    def __init__(self, manager, args):
+    def __init__(self, manager, args, config):
         super(Fetcher, self).__init__(manager)
         self.device = None
         self.timestamps = None
         address = args.address
         index = args.index
 
+        if not address in config:
+            config[address] = {}
```
Comment (bentiss) [inline tools/kete.py]: `tuhi` has per BT address directories with `settings.ini` instead of one gigantic `settings.ini`. Not sure which one is the best given the few options we have to configure

File: tools/kete.py [hunk 4]
```diff
@@ -586,19 +592,63 @@ def run(self):
...
+                elif self.orientation == 2:
+                    x, y = height - y, x
+                elif self.orientation == 3:
+                    x, y = width - x, height - y
+
+                delta = (p['pressure'] - 1000.0)/ 1000.0
+                stoke_width = 0.4 + 0.20 * delta
+                points_with_sk_width.append((x, y, stoke_width))
+
+            if self.handle_pressure:
```
Comment (bentiss) [inline tools/kete.py]: this part should be in its own separate patch

File: tools/kete.py [hunk 5]
```diff
@@ -586,19 +592,63 @@ def run(self):
     def json_to_svg(self, js, filename):
         dimensions = js['dimensions']
         if dimensions == [0, 0]:
-            dimensions = 100, 100
-        svg = svgwrite.Drawing(filename=filename, size=dimensions)
+            width, height = 100, 100
+        else:
+            # Original diemnsions are too big for SVG Standard
+            # so we nomalize them
+            width, height = dimensions[0]/100, dimensions[1]/100
```
Comment (bentiss) [inline tools/kete.py]: This should be in its own separate patch

File: tools/kete.py [hunk 6]
```diff
@@ -706,6 +756,10 @@ def __init__(self, completekey='tab', stdin=None, stdout=None):
         except FileExistsError:
             pass
 
+        self._config_file = os.path.join(CONFIG_PATH, 'settings.ini')
+        self._config = configparser.ConfigParser()
+        self._config.read(self._config_file)
```
Comment (bentiss) [inline tools/kete.py]: I wonder if we should not create a basic `settings.ini` when opening `kete` so we can document the various values that we are going to use.

Something like `if not os.path.exists(self._config_file)`

**Review by bentiss — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution. I am generally happy with the changes, but I think I am going to request some changes to make the PR easier to read in the future:

- please split the patch in 3:
  - one first patch to normalize the SVG dimensions
  - then one patch that adds the `settings.ini` and sets the correct orientation
  - and finally one patch to handle the variable tablet pressure

Then, in each patch please take into account the various comments inlined.

File: tools/kete.py (hunk 3 above)
Comment (mont5piques) [reply tools/kete.py]: I think most people have at most 2 tablets. I don't see the interest of creating multiple directory for a not so common use-case. What do you think ?

**Comment by mont5piques** [comment]
I'll make the requested changes and post a new PR.

**Comment by bentiss** [comment]
> I'll make the requested changes and post a new PR.

Thanks, but no need to create a new PR. Just force push on your branch on your repo and github will happily take the new branch into account.

File: tools/kete.py (hunk 3 above)
Comment (bentiss) [reply tools/kete.py]: Problem is, I do not recall the rational behind the separate dir for tablets in tuhi. I guess we can keep the single file for now and decide what to do before making a new release.

At least we can get @whot's opinion when he is back.
